erpnext/stock/doctype/batch/batch.py

```diff
@@ -362,7 +362,13 @@ def get_sufficient_batch_or_fifo(item_code, warehouse, qty=1.0, conversion_facto
 
 	for batch in batches:
 		if remaining_stock_qty <= 0:
-			break
+			selected_batches.append(frappe._dict({
```

Comment:
Gotta make this conditional since other features rely on this function to return only selected batches. Argument include_all_batches or include_unselected_batches